src/cmd2.cc

```diff
@@ -556,58 +545,9 @@ void do_cmd_go_down(void)
 
 		/* Leaving */
 		p_ptr->leaving = TRUE;
-
-		if (!fall_trap)
-		{
```

Comment:
AFAICT this block should not be removed; "fall_trap" was false by default, and !false == true.